lib/puck/jar.rb

```diff
@@ -140,6 +148,14 @@ def create
                 zipfileset dir: spec[:base_path], prefix: File.join(JAR_GEM_HOME, spec[:versioned_name])
               end
             end
+
+            merge_archives.each do |archive, target_path|
+              if target_path
+                zipfileset src: archive, prefix: target_path, excludes: SIGNATURE_FILES
+              else
+                zipfileset src: archive, excludes: SIGNATURE_FILES
+              end
+            end
```

Comment:
I've done an attempt of rewriting this. I'm not certain that it simplified anything, but now they are handled the same.